bindings/python/iree/runtime/vm.cc

```diff
@@ -291,6 +291,15 @@ void VmVariantList::PushBufferView(HalDevice& device,
                  "Error moving buffer view");
 }
 
+VmVariantList VmVariantList::GetAsList(int index) {
+  iree_vm_variant_t v = iree_vm_variant_empty();
+  CheckApiStatus(iree_vm_list_get_variant(raw_ptr(), index, &v),
+                 "Could not access list element");
+  iree_vm_list_t* sub_list = iree_vm_list_deref(v.ref);
```

Comment:
this can return null if the given variant is not a list - you could either check that and bail here or change all of this to:
```c
iree_vm_list_t* sub_list = iree_vm_list_get_ref_deref(raw_ptr(), index, iree_vm_list_get_descriptor());
if (!sub_list) ...fail
```

if you want the best error messages, though, this is probably best:
```c
iree_vm_ref_t ref = {0};
CheckApiStatus(iree_vm_list_get_ref_assign(raw_ptr(), index, &ref));
iree_vm_list_t* sub_list = NULL;
CheckApiStatus(iree_vm_list_check_deref(ref, &sub_list));
```
(that gives you out of bounds, null elements, and wrong types errors)

Reply:
Thanks, implemented the latter and added unit tests.